autoload/nvimdev.vim

```diff
@@ -42,18 +42,30 @@ function! nvimdev#init(path) abort
     endif
   endif
 
+  let do_bufenter = 0
+
   augroup nvimdev
     autocmd!
     autocmd BufRead,BufNewFile *.h set filetype=c
     if get(g:, 'nvimdev_auto_ctags', 0) || get(g:, 'nvimdev_auto_cscope', 0)
       autocmd BufWritePost *.c,*.h,*.lua call s:build_db()
     endif
     if get(g:, 'nvimdev_build_readonly', 1)
-      execute 'autocmd BufRead ' . s:path . '/build/* setlocal readonly nomodifiable'
-      execute 'autocmd BufRead ' . s:path . '/.deps/* setlocal readonly nomodifiable'
+      let glob = fnameescape(s:path).'/{build,.deps}/*'
+      execute 'autocmd BufEnter '.glob.' '
+            \ .(has('nvim-0.4.0') ? '++once ' : '')
+            \ .'setlocal readonly nomodifiable'
+      if expand('%:p') =~# glob2regpat(glob)
+        let do_bufenter = 1
+      endif
```

Comment:
This is done to avoid "No matching autocommands", and I do not like using "silent" to avoid this.

Reply:
what was wrong with `exists('#nvimdev#BufEnter')`?

Reply:
It will cause the message..
The event exists, but does not match the buffer then..

Reply:
https://github.com/vim/vim/issues/4300